src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DoNotIgnoreReturnValue.cs

```diff
@@ -15,61 +16,82 @@ namespace Microsoft.NetCore.Analyzers.Runtime
-            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueTitle)),
+            s_title,
             CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueMessage)),
             DiagnosticCategory.Reliability,
             RuleLevel.BuildWarning,
-            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueDescription)),
+            s_description,
             isPortedFxCopRule: false,
             isDataflowRule: false);
 
         internal static readonly DiagnosticDescriptor DoNotIgnoreReturnValueRuleWithMessage = DiagnosticDescriptorHelper.Create(
             CA2022RuleId,
-            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueTitle)),
+            s_title,
             CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueMessageCustom)),
             DiagnosticCategory.Reliability,
             RuleLevel.BuildWarning,
-            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueDescription)),
+            s_description,
             isPortedFxCopRule: false,
             isDataflowRule: false);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(DoNotIgnoreReturnValueRule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(DoNotIgnoreReturnValueRule, DoNotIgnoreReturnValueRuleWithMessage);
 
         public override void Initialize(AnalysisContext context)
         {
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            context.RegisterOperationBlockStartAction(context =>
+            context.RegisterCompilationStartAction(context =>
             {
                 if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemDiagnosticsCodeAnalysisDoNotIgnoreAttribute, out var doNotIgnoreAttribute))
                 {
                     return;
                 }
 
+                context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksTask1, out var taskOfT);
+
                 context.RegisterOperationAction(context =>
                 {
-                    var invocation = (IInvocationOperation)context.Operation;
+                    var callSite = context.Operation;
+                    var method = ((IInvocationOperation)callSite).TargetMethod;
+
+                    // It would be an authoring error, but ensure the method returns a value
+                    if (method.ReturnsVoid)
+                    {
+                        return;
+                    }
+
+                    // If the call is awaited, and the method returns a Task<T>, then we'll check the await result for consumption
+                    if (taskOfT is not null && callSite.Parent?.Kind is OperationKind.Await && (method.ReturnType as INamedTypeSymbol)?.ConstructedFrom == taskOfT)
```

Comment:
- I think better to avoid `==` (it's not overloaded in symbols AFAIK). Let's replace with `taskOfT.Equals(...., SymbolEqualityComparer.Default)`
- `ValueTask<T>`

Reply:
Thanks for these notes. And really, this needs to handle all task-like objects too. I'm changing the approach...